### PR TITLE
feat: Add IPv4 CIDR and IPv6 CIDR validations

### PR DESCRIPTION
**Pull request by langscot**
Implements #342. Adds separate validations for both IPv4 and IPv6 CIDR notations.

### CODE REVIEW

**Comment by fabian-hiller** [comment]
Thank you for your contribution. I will review this PR in 1 or 2 weeks.

File: library/src/regex.ts [hunk 1]
```diff
@@ -47,12 +47,26 @@ export const IPV4_REGEX =
...
+ */
+export const IPV6_CIDR_REGEX =
+  // eslint-disable-next-line redos-detector/no-unsafe-regex, regexp/no-super-linear-backtracking -- false positive
+  /^s*((([\dA-Fa-f]{1,4}:){7}([\dA-Fa-f]{1,4}|:))|(([\dA-Fa-f]{1,4}:){6}(:[\dA-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([\dA-Fa-f]{1,4}:){5}(((:[\dA-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([\dA-Fa-f]{1,4}:){4}(((:[\dA-Fa-f]{1,4}){1,3})|((:[\dA-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([\dA-Fa-f]{1,4}:){3}(((:[\dA-Fa-f]{1,4}){1,4})|((:[\dA-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([\dA-Fa-f]{1,4}:){2}(((:[\dA-Fa-f]{1,4}){1,5})|((:[\dA-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([\dA-Fa-f]{1,4}:)(((:[\dA-Fa-f]{1,4}){1,6})|((:[\dA-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:[\dA-Fa-f]{1,4}){1,7})|((:[A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)*(\/(\d|[1-9]\d|1[01]\d|12[0-8]))$/u;
```
Comment (kurtextrem) [inline library/src/regex.ts:68]: I don't fully understand why the `s*` is needed here. What does it do?

Comment (fabian-hiller) [reply library/src/regex.ts:68]: Thank you for the review!

Comment (ariskemper) [reply library/src/regex.ts:68]: @kurtextrem thanks for the review, i completly agree.

**Comment by fabian-hiller** [comment]
Since we have rewritten the entire library, this PR is unfortunately outdated and cannot be merged. Is anyone interested in updating this PR or creating a new one?

**Comment by fabian-hiller** [comment]
Unfortunately, due to our rewrite a few months ago, this PR is no longer up to date with our source code. I still plan to add this action, but other things have a higher priority for me at the moment. If you or anyone else in the community is interested in creating a new PR, that would be great.

**Pull request closed without merging**